src/main/java/com/synopsys/integration/alert/channel/jira/server/util/JiraServerIssueHandler.java

```diff
@@ -0,0 +1,104 @@
+import com.synopsys.integration.alert.common.message.model.ComponentItem;
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jira.common.model.request.IssueCommentRequestModel;
+import com.synopsys.integration.jira.common.model.request.builder.IssueRequestModelFieldsMapBuilder;
+import com.synopsys.integration.jira.common.model.response.IssueResponseModel;
+import com.synopsys.integration.jira.common.server.model.IssueCreationRequestModel;
+import com.synopsys.integration.jira.common.server.model.IssueSearchIssueComponent;
+import com.synopsys.integration.jira.common.server.model.IssueSearchResponseModel;
+import com.synopsys.integration.jira.common.server.service.IssueService;
+
+public class JiraServerIssueHandler extends JiraIssueHandler {
+    private static final Logger logger = LoggerFactory.getLogger(JiraServerIssueHandler.class);
+
+    private final IssueService issueService;
+    private final JiraServerProperties jiraProperties;
+
+    private final JiraServerIssuePropertyHandler jiraIssuePropertyHelper;
+
+    public JiraServerIssueHandler(IssueService issueService, JiraServerProperties jiraProperties, JiraMessageParser jiraMessageParser, Gson gson, JiraServerTransitionHandler jiraTransitionHandler,
+        JiraServerIssuePropertyHandler jiraIssuePropertyHandler) {
+        super(jiraMessageParser, gson, jiraTransitionHandler, jiraIssuePropertyHandler);
+        this.issueService = issueService;
+        this.jiraProperties = jiraProperties;
+        this.jiraIssuePropertyHelper = jiraIssuePropertyHandler;
+    }
+
+    @Override
+    public IssueResponseModel createIssue(String issueCreator, String issueType, String projectName, IssueRequestModelFieldsMapBuilder fieldsBuilder) throws IntegrationException {
+        return issueService.createIssue(new IssueCreationRequestModel(issueCreator, issueType, projectName, fieldsBuilder));
+    }
+
+    @Override
+    protected List<IssueResponseModel> retrieveExistingIssues(String projectKey, String provider, LinkableItem topic, LinkableItem nullableSubTopic, ComponentItem componentItem, String alertIssueUniqueId)
+        throws IntegrationException {
+        List<IssueSearchIssueComponent> searchIssueModels = jiraIssuePropertyHelper
+                                                                .findIssues(projectKey, provider, topic, nullableSubTopic, componentItem, alertIssueUniqueId)
+                                                                .map(IssueSearchResponseModel::getIssues)
+                                                                .orElse(List.of());
+        List<IssueResponseModel> issues = new LinkedList<>();
+
+        for (IssueSearchIssueComponent searchIssue : searchIssueModels) {
+            try {
+                issues.add(issueService.getIssue(searchIssue.getKey()));
```

Comment:
Is an additional request required here? I thought the IssueSearchComponent had a list of issues.

Reply:
It does for Jira Cloud for Jira server it has a different object that does not contain the same object fields.  So to return a list of the common objects IssueResponseModel that the retrieveExsitingIssues method signature needs, this additional request is made to get the same type of objects from Jira Server and Jira Cloud.